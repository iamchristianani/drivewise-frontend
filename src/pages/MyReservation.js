import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import './style/MyReservation.css';
import { BiLeftArrow } from 'react-icons/bi';
import { BsBookmarks } from 'react-icons/bs';
import Navbar from '../components/Navbar';
import { getReservationsAction } from '../redux/reservations/reservations';
import { fetchCars } from '../redux/cars/cars';
import ReservationsTable from '../components/ReservationsTable';

const MyReservation = () => {
  const dispatch = useDispatch();
  const reservations = useSelector((state) => state.reservations);
  const cars = useSelector((state) => state.cars);
  const { user } = useSelector((state) => state.authentications);

  useEffect(() => {
    dispatch(fetchCars());
    dispatch(getReservationsAction(user.id));
  }, [dispatch]);

  const getCarName = (carId) => {
    const car = cars.cars.find((c) => c.id === carId);
    return car ? car.make : 'Unknown Car';
  };

  return (
    <div className="page_container">
      <Navbar />
      <div className="details_container">
        <div className="inner-details">
          <div className="home_header">
            <h1>MY RESERVATIONS</h1>
            <p>Showing all your test drive reservations</p>
            <div className="dot_divider">.............................</div>
          </div>
          <div className="reservation-box">
            <table className="reservation-table">
              <thead>
                <tr>
                  <th>Car Name</th>
                  <th>City</th>
                  <th>Reservation Date</th>
                </tr>
              </thead>
              <tbody>
                {
                  reservations.length ? (
                    reservations.map((item) => (
                      <ReservationsTable
                        key={item.id}
                        id={item.id}
                        carName={getCarName(item.car_id)}
                        city={item.city}
                        reservationDate={item.reservation_date}
                      />
                    ))
                  ) : (
                    <tr className="no-data-box">
                      <td colSpan="3" className="no-data-td">
                        <BsBookmarks className="no-data-icon" />
                        <br />
                        No reservations available
                      </td>
                    </tr>
                  )
                }
              </tbody>
            </table>
          </div>
        </div>
        <div className="back-button">
          <NavLink to="/">
            <button type="button" className="scroll left-scroll" aria-label="Scroll Left"><BiLeftArrow /></button>
          </NavLink>
        </div>
      </div>
    </div>
  );
};

export default MyReservation;
