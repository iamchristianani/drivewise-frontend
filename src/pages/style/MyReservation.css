.reservation-box {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: 20px 0;
}

.reservation-table {
  border-collapse: separate;
  border-spacing: 0;
  width: 100%;
}

.reservation-table th,
.reservation-table td {
  padding: 8px;
  text-align: center;
  border: 1px solid black;
}

.reservation-table th {
  background-color: #97bf0f;
  color: white;
  font-weight: bold;
  padding: 15px;
}

.reservation-table tr:nth-child(even) {
  background-color: #f2f2f2;
}

.reservation-table tr:hover {
  background-color: #ddd;
}
