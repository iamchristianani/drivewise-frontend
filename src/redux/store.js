import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import carsReducer from './cars/cars';
import detailsReducer from './cars/carDetails';
import reservationsReducer from './reservations/reservations';
import authSlice from './loginByUsername';

const store = configureStore(
  {
    reducer: {
      authentications: authSlice,
      cars: carsReducer,
      details: detailsReducer,
      reservations: reservationsReducer,
    },
  },
  applyMiddleware(thunk),
);

export default store;
