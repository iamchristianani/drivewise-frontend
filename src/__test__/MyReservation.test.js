import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import MyReservation from '../pages/MyReservation';

describe('Home Test', () => {
  test('to render all car tests reservations: ', () => {
    const home = render(
      <Provider store={store}>
        <BrowserRouter>
          <MyReservation />
        </BrowserRouter>
      </Provider>,
    );
    expect(home).toMatchSnapshot();
  });
});
