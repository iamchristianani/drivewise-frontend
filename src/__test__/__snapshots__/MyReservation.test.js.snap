// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test to render all car tests reservations:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page_container"
      >
        <nav>
          <a
            href="/"
          >
            <img
              alt="Drivewise"
              src="drivewise_logo.png"
            />
          </a>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-lg "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-x not-active"
            data-icon="x"
            data-prefix="fas"
            data-testid="mobile-button"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
              fill="currentColor"
            />
          </svg>
          <ul
            class="mobile-menu "
            data-testid="mobile-menu"
          >
            <li
              class=""
            >
              <a
                href="/"
              >
                Cars
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/reservation_form"
              >
                Add Reservation
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/my_reservations"
              >
                My Reservations
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/new_car"
              >
                Add Car
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/delete_car"
              >
                Delete Car
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="details_container"
        >
          <div
            class="inner-details"
          >
            <div
              class="home_header"
            >
              <h1>
                MY RESERVATIONS
              </h1>
              <p>
                Showing all your test drive reservations
              </p>
              <div
                class="dot_divider"
              >
                .............................
              </div>
            </div>
            <div
              class="reservation-box"
            >
              <table
                class="reservation-table"
              >
                <thead>
                  <tr>
                    <th>
                      Car Name
                    </th>
                    <th>
                      City
                    </th>
                    <th>
                      Reservation Date
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    id="reservation-2"
                  >
                    <td>
                      Unknown Car
                    </td>
                    <td>
                      Lagos
                    </td>
                    <td>
                      2023-05-08
                    </td>
                  </tr>
                  <tr
                    id="reservation-6"
                  >
                    <td>
                      Unknown Car
                    </td>
                    <td>
                      Johannesburg
                    </td>
                    <td>
                      2023-05-12
                    </td>
                  </tr>
                  <tr
                    id="reservation-10"
                  >
                    <td>
                      Unknown Car
                    </td>
                    <td>
                      Baghdad
                    </td>
                    <td>
                      2023-05-16
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="back-button"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <button
                aria-label="Scroll Left"
                class="scroll left-scroll"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
                  />
                </svg>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page_container"
    >
      <nav>
        <a
          href="/"
        >
          <img
            alt="Drivewise"
            src="drivewise_logo.png"
          />
        </a>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-lg "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-x not-active"
          data-icon="x"
          data-prefix="fas"
          data-testid="mobile-button"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
            fill="currentColor"
          />
        </svg>
        <ul
          class="mobile-menu "
          data-testid="mobile-menu"
        >
          <li
            class=""
          >
            <a
              href="/"
            >
              Cars
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/reservation_form"
            >
              Add Reservation
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/my_reservations"
            >
              My Reservations
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/new_car"
            >
              Add Car
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/delete_car"
            >
              Delete Car
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="details_container"
      >
        <div
          class="inner-details"
        >
          <div
            class="home_header"
          >
            <h1>
              MY RESERVATIONS
            </h1>
            <p>
              Showing all your test drive reservations
            </p>
            <div
              class="dot_divider"
            >
              .............................
            </div>
          </div>
          <div
            class="reservation-box"
          >
            <table
              class="reservation-table"
            >
              <thead>
                <tr>
                  <th>
                    Car Name
                  </th>
                  <th>
                    City
                  </th>
                  <th>
                    Reservation Date
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  id="reservation-2"
                >
                  <td>
                    Unknown Car
                  </td>
                  <td>
                    Lagos
                  </td>
                  <td>
                    2023-05-08
                  </td>
                </tr>
                <tr
                  id="reservation-6"
                >
                  <td>
                    Unknown Car
                  </td>
                  <td>
                    Johannesburg
                  </td>
                  <td>
                    2023-05-12
                  </td>
                </tr>
                <tr
                  id="reservation-10"
                >
                  <td>
                    Unknown Car
                  </td>
                  <td>
                    Baghdad
                  </td>
                  <td>
                    2023-05-16
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="back-button"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <button
              aria-label="Scroll Left"
              class="scroll left-scroll"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
